### ISSUE TITLE
JUnit report is not indented

### ISSUE THREAD

**Issue by AirQuick** [xspec/xspec#226]
JUnit report XML file is not indented. For example, see [this](https://raw.githubusercontent.com/xspec/xspec/68086d67ee5d721e2878a0bb815321cd0f92e973/test/end-to-end/cases/expected/schematron-01-junit-norm.xml).
That makes it hard to visually inspect the JUnit report files.

JUnit report files are generated by `junit-report.xsl`. Although it has `xsl:output` with `indent="yes"`, the `xsl:output` is not used (because of unreferenced `name="escaped"`):
https://github.com/xspec/xspec/blob/4a9c470421c671fbb32a94d76153ff3405b41a26/src/reporter/junit-report.xsl#L28

**Comment by AirQuick** [comment]
@cirulls ,
Looks like the JUnit report was implemented in #6. Do you know these?
* The reason the output XML is not indented
* The purpose of the unreferenced `xsl:output`
* How [Kal Ahmed](https://github.com/xspec/xspec/blob/4a9c470421c671fbb32a94d76153ff3405b41a26/src/reporter/junit-report.xsl#L13) was involved. (just curious)

**Comment by cirulls** [comment]
Yes, I implemented this back in 2016, it was one of my first contributions to XSpec. There are more technical notes in #2. Regarding your questions:

1. It was probably accidental or maybe for space reason as the JUnit is quite verbose. I agree that the indentation would be better for readability and debugging purposes.
2. I'll check the unreferenced name attribute in `xsl:output`, if that's the case there is certainly a contradiction. I'll get back to you.
3. @kal is a colleague of mine. At the time we independently wrote JUnit serializations for XSpec  and I collected our separated efforts in that pull request (this may well be the explanation for the unreferenced attribute).
